helm/dagster/schema/schema_tests/test_instance.py

```diff
@@ -663,3 +665,30 @@ def test_run_coordinator_has_schema(json_schema_model, run_coordinator_class):
     run_coordinator_fields = set(map(to_camel_case, run_coordinator_class.config_type().keys()))
 
     assert json_schema_fields == run_coordinator_fields
+
+
+def test_retention(template: HelmTemplate):
```

Comment:
add test for backcompat (e.g. with default Helm values, this field is not included in `instance.yaml`)